newrelic/structures_newrelic_dashboard.go

```diff
@@ -6,22 +6,22 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 
-	newrelic "github.com/paultyng/go-newrelic/v4/api"
+	"github.com/newrelic/newrelic-client-go/pkg/dashboards"
 )
 
-// Assemble the *newrelic.Dashboard struct.
+// Assemble the *dashboards.Dashboard struct.
 // Used by the newrelic_dashboard Create and Update functions.
-func expandDashboard(d *schema.ResourceData) (*newrelic.Dashboard, error) {
-	metadata := newrelic.DashboardMetadata{
+func expandDashboard(d *schema.ResourceData) (*dashboards.Dashboard, error) {
+	metadata := dashboards.DashboardMetadata{
 		Version: 1,
 	}
 
-	dashboard := newrelic.Dashboard{
+	dashboard := dashboards.Dashboard{
 		Title:      d.Get("title").(string),
 		Metadata:   metadata,
 		Icon:       d.Get("icon").(string),
-		Visibility: d.Get("visibility").(string),
-		Editable:   d.Get("editable").(string),
+		Visibility: dashboards.VisibilityType(d.Get("visibility").(string)),
```

Comment:
Is this redundant? Or is there an advantage to doing it this way?

Reply:
The types are now different than `string`, this conversion is necessary for the package to compile.